Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/EyeTrackingTarget.cs

```diff
@@ -3,69 +3,186 @@
 
 using UnityEngine;
 using UnityEngine.Events;
+using Microsoft.MixedReality.Toolkit.Input;
+using System;
 
-namespace Microsoft.MixedReality.Toolkit.Input
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos.EyeTracking
 {
     /// <summary>
-    /// A game object with the "Target" script attached can be selected and may react to hover states.
+    /// A game object with the "EyeTrackingTarget" script attached reacts to being looked at independent of other available inputs.
     /// </summary>
-    public class EyeTrackingTarget : BaseEyeFocusHandler, IMixedRealityPointerHandler, IMixedRealitySpeechHandler
+    [RequireComponent(typeof(EyeTrackingTarget))]
+    public class EyeTrackingTarget : InputSystemGlobalListener, IMixedRealityPointerHandler, IMixedRealitySpeechHandler
     {
         [SerializeField]
         private MixedRealityInputAction selectAction = MixedRealityInputAction.None;
 
         [SerializeField]
-        private MixedRealityInputAction[] voice_select = null;
+        private MixedRealityInputAction[] voiceSelect = null;
 
-        [Tooltip("If true, the game object is set to the center of the currently looked at target.")]
-        public bool cursor_snapToCenter = false;
+        public bool eyeCursorSnapToTargetCenter = false;
 
-        [Tooltip("Event is triggered when the user starts to look at the target.")]
+        [Tooltip("Event to be triggered when the looked at target is selected.")]
+        [Range(0, 10)]
         [SerializeField]
-        private UnityEvent OnLookAtStart = null;
+        private float dwellTimeInSec = 0.8f;
+
+        [Tooltip("Event is triggered when the user starts to look at the target.")]        
+        public UnityEvent OnLookAtStart = null;
 
         [Tooltip("Event is triggered when the user continues to look at the target.")]
-        [SerializeField]
-        private UnityEvent WhileLookingAtTarget = null;
+        public UnityEvent WhileLookingAtTarget = null;
 
         [Tooltip("Event to be triggered when the user is looking away from the target.")]
-        [SerializeField]
-        private UnityEvent OnLookAway = null;
+        public UnityEvent OnLookAway = null;
 
         [Tooltip("Event to be triggered when the target has been looked at for a given predefined duration.")]
-        [SerializeField]
-        private UnityEvent OnDwell = null;
+        public UnityEvent OnDwell = null;
 
         [Tooltip("Event to be triggered when the looked at target is selected.")]
-        [SerializeField]
-        private UnityEvent OnSelected = null;
+        public UnityEvent OnSelected = null;
+        
+        public bool IsLookedAt { get; private set; }
```

Comment:
Nit: Add a comment for public properties (in particular, is this true if you're just looking at it, or does it wait for some little amount of dwell as well? i.e. if my eye just flickers over it I assume this will return yes very transiently)

Reply:
Updated!